pkg/local_object_storage/metabase/version.go

```diff
@@ -12,14 +12,25 @@ const version = 2
 
 var versionKey = []byte("version")
 
+// ErrOutdatedVersion is returned on initializing
+// an existing metabase that is not compatible with
+// the current code version.
+type ErrOutdatedVersion struct {
+	stored uint64
+}
+
+func (e ErrOutdatedVersion) Error() string {
+	return fmt.Sprintf("invalid version: expected=%d, stored=%d", version, e.stored)
```

Comment:
I don't like complex error types, what about `var ErrOutdatedVersion = errors.New("invalid version")` + `return fmt.Errorf("%w: expected...", ErrOutdatedVersion, ...)`?

Reply:
Here I agree with @fyrchik at least since we don't decode the error structure. We could use `errors.Is`.